agent.go

```diff
@@ -120,15 +120,16 @@ type container struct {
 }
 
 type pod struct {
-	id         string
-	running    bool
-	containers map[string]*container
-	ctl        *os.File
-	tty        *os.File
-	stdinList  map[uint64]*os.File
-	network    hyper.Network
-	stdinLock  sync.Mutex
-	ttyLock    sync.Mutex
+	id             string
+	running        bool
+	containers     map[string]*container
+	ctl            *os.File
+	tty            *os.File
+	stdinList      map[uint64]*os.File
+	network        hyper.Network
+	stdinLock      sync.Mutex
+	ttyLock        sync.Mutex
+	containersLock sync.RWMutex
```

Comment:
@sboeuf This only protects the containers map, so I have used the name containersLock here.